modules/loader/src/main/java/com/opengamma/strata/loader/csv/CsvInfoResolver.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.loader.csv;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.collect.io.CsvRow;
+import com.opengamma.strata.product.PositionInfoBuilder;
+import com.opengamma.strata.product.TradeInfoBuilder;
+import com.opengamma.strata.product.common.ExchangeId;
+import com.opengamma.strata.product.etd.EtdContractCode;
+import com.opengamma.strata.product.etd.EtdContractSpec;
+import com.opengamma.strata.product.etd.EtdContractSpecId;
+import com.opengamma.strata.product.etd.EtdIdUtils;
+import com.opengamma.strata.product.etd.EtdType;
+
+/**
+ * Resolves security information from CSV files, enriching the parser.
+ * <p>
+ * Data loaded from a CSV may contain additional information that needs to be captured.
+ * This plugin point allows the additional CSV columns to be parsed and captured.
+ */
+public interface CsvInfoResolver {
+
+  /**
+   * Obtains an instance that uses the standard set of reference data.
+   * 
+   * @return the loader
+   */
+  public static CsvInfoResolver standard() {
+    return StandardCsvInfoResolver.of(ReferenceData.standard());
+  }
+
+  /**
+   * Obtains an instance that uses the specified set of reference data.
+   * 
+   * @param refData  the reference data
+   * @return the loader
+   */
+  public static CsvInfoResolver of(ReferenceData refData) {
+    return StandardCsvInfoResolver.of(refData);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the reference data being used.
+   * 
+   * @return the reference data
+   */
+  public abstract ReferenceData getReferenceData();
+
+  /**
+   * Parses attributes into {@code TradeInfo}.
+   * 
+   * @param row  the CSV row to parse
+   * @param builder  the builder to update
+   */
+  public default void parseTradeAttributes(CsvRow row, TradeInfoBuilder builder) {
+    // do nothing
+  }
+
+  /**
+   * Parses attributes into {@code PositionInfo}.
+   * 
+   * @param row  the CSV row to parse
+   * @param builder  the builder to update
+   */
+  public default void parsePositionAttributes(CsvRow row, PositionInfoBuilder builder) {
+    // do nothing
```

Comment:
Given that these are builder mutating methods, 'populate' might be a better name than 'parse'.

Are we assuming that overrides of these methods may set non-attribute values on the builder (e.g. settlement date, trade date)? 
If yes then parse*Info would be a better name than parse*Attributes 

If the methods are only expected to populate attributes then we could return a map of attributes and allow the clients to populate the builder